src/Controls/tests/TestCases.HostApp/Issues/Issue26598.cs

```diff
@@ -0,0 +1,136 @@
+using Microsoft.Maui.Controls;
+
+namespace Maui.Controls.Sample.Issues;
+[Issue(IssueTracker.Github, 26598, "Tabbar disappears when navigating back from page with hidden TabBar in iOS", PlatformAffected.iOS)]
+public class Issue26598 : TestShell
+{
+	TabBar tabBar = new TabBar();
+
+	// Create first ShellContent
+	ShellContent homeShellContent = new ShellContent
+	{
+		ContentTemplate = new DataTemplate(() => new Issue26598Home()),
+		Title = "Home",
+		AutomationId = "Issue26598Home",
+		Route = nameof(Issue26598Home)
+	};
+
+	// Create second ShellContent
+	ShellContent recentShellContent = new ShellContent
+	{
+		ContentTemplate = new DataTemplate(() => new Issue26598Recent()),
+		Title = "Recent",
+		Route = nameof(Issue26598Recent)
+	};
+
+	protected override void Init()
+	{
+		Routing.RegisterRoute("Issue26598Inner", typeof(Issue26598Inner));
+		Routing.RegisterRoute(nameof(Issue26589NonTab), typeof(Issue26589NonTab));
+		tabBar.Items.Add(homeShellContent);
+		tabBar.Items.Add(recentShellContent);
+		Items.Add(tabBar);
+		Shell.SetTabBarIsVisible(this, true);
+	}
+}
+
+public class Issue26598Home : ContentPage
```

Comment:
Can you move all these classes to be inner classes of `Issue26598` so they don't pollute the namespace?

Reply:
Yes @PureWeen! I have refactored the code by moving all the classes inside `Issue26598` as inner classes.